flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -550,6 +555,7 @@ private void checkResourceRequirements() {
                         jobId, resourceTracker.getAcquiredResources(jobId));
             }
         }
+        requirementsCheckFuture.complete(null);
```

Comment:
I'd suggest to deduplicate this by moving it to `checkResourceRequirementsWithDelay`, after calling `checkResourceRequirements`.

Reply:
And maybe add a null check for `requirementsCheckFuture`.